src/api/api.cpp

```diff
@@ -611,16 +629,16 @@ extern "C" migraphx_status migraphx_onnx_options_create(migraphx_onnx_options_t*
     });
 }
 
-extern "C" migraphx_status
-migraphx_onnx_options_set_input_parameter_shape(migraphx_onnx_options_t onnx_options,
-                                                const char* name,
-                                                const size_t* dims,
-                                                const size_t dim_num)
+extern "C" migraphx_status migraphx_onnx_options_set_input_parameter_shape(
+    migraphx_onnx_options_t onnx_options, const char* name, size_t* dims, size_t dims_size)
 {
     return migraphx::try_([&] {
         if(onnx_options == nullptr)
             MIGRAPHX_THROW(migraphx_status_bad_param, "Bad parameter onnx_options: Null pointer");
-        migraphx::set_input_parameter_shape((onnx_options->object), (name), (dims), (dim_num));
+        if(dims == nullptr)
+            MIGRAPHX_THROW(migraphx_status_bad_param, "Bad parameter dims: Null pointer");
+        migraphx::set_input_parameter_shape(
+            (onnx_options->object), (name), (std::vector<size_t>(dims, dims + dims_size)));
```

Comment:
This change revert to the code in my previous PR parse_onnx_using_explicit_input_shape. The problem here is that if dim is nullptr, an exception is thrown.

But in the function onnx_options::set_input_parameter_shape(name, dims.data(), dim.size()) in migraphx.hpp, if dim is empty (used to represent a scalar shape in both numpy and ort), we will throw an exception.
That is the reason I made some changes in my PR python_scalar_input_shape.

Reply:
Sorry, I am not sure why this was reverted, maybe a merge mistake. I will try to fix it.